### PR TITLE
Repo rename changes

### PR DESCRIPTION
**Pull request by sagidayan**
Not including:
 - example readme files
 - docs

These will be added in separate PR's due to the size of this PR.

### CODE REVIEW

**Comment by sagidayan** [comment]
/hold

This is WIP. Changed from "ocm" to "redhatcs". If accepted - we should also update the examples and ci.

File: build/ci-tf-bin.Dockerfile [hunk 1]
```diff
@@ -6,13 +6,13 @@ RUN microdnf update -y && \
     microdnf install -y git make go-toolset && \
     microdnf clean all
 
-# terraform-provider-ocm repo
-COPY . ./terraform-provider-ocm
+# terraform-provider-red-hat-cloud-services repo
+COPY . ./terraform-provider-red-hat-cloud-services
```
Comment (bardielle) [inline build/ci-tf-bin.Dockerfile:10]: @enriquebelarte we also need a PR for openshift/release repo, doesn't we?

File: ci/e2e/terraform_provider_ocm_fullcycle_test.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ import (
 	client "github.com/openshift-online/ocm-sdk-go"
 	"github.com/openshift-online/ocm-sdk-go/logging"
 	"github.com/segmentio/ksuid"
-	"github.com/terraform-redhat/terraform-provider-ocm/ci/helper"
+	"github.com/terraform-redhat/terraform-provider-rhcs/ci/helper"
```
Comment (bardielle) [inline ci/e2e/terraform_provider_ocm_fullcycle_test.go:22]: Please also change the const name (line 28) terraformProviderOCMFilesDir -> terraformProviderRhcsFilesDir
"terraform_provider_ocm_files" -> "terraform_provider_rhcs_files"

**Review by bardielle — COMMENTED** [review]
I didn't see that you change the docs (run make generate command)

File: build/ci-tf-bin.Dockerfile (hunk 1 above)
Comment (sagidayan) [reply build/ci-tf-bin.Dockerfile:10]: Updated this to `rhcs` as well...

We have a PR  https://github.com/openshift/release/pull/40677

File: .github/workflows/publish-release.yaml [hunk 3]
```diff
@@ -39,7 +39,7 @@ jobs:
       uses: actions/checkout@v2
 
     - name: Workaround to fix git dir
-      run: git config --system --add safe.directory /__w/terraform-provider-ocm/terraform-provider-ocm
+      run: git config --system --add safe.directory /__w/terraform-provider-red-hat-cloud-services/terraform-provider-red-hat-cloud-services
```
Comment (nirarg) [inline .github/workflows/publish-release.yaml:42]: rhcs

File: README.md [hunk 4]
```diff
@@ -51,10 +51,10 @@ The following items are limitations with the current release of the OCM Terrafor
 
 The example Terraform files are all considered in development:
 ### Prior to creating clusters
-1. [Account Roles terraform](https://github.com/terraform-redhat/terraform-provider-ocm/blob/a42779d6b6712f4dde358344f44b782e4dfcd120/examples/create_rosa_cluster/create_rosa_sts_cluster/classic_sts/account_roles/README.md)
+1. [Account Roles terraform](https://github.com/terraform-redhat/terraform-provider-rhcs/blob/a42779d6b6712f4dde358344f44b782e4dfcd120/examples/create_rosa_cluster/create_rosa_sts_cluster/classic_sts/account_roles/README.md)
```
Comment (nirarg) [inline README.md:54]: Lets point to main branch here and not hash

**Review by bardielle — APPROVED** [review]
/lgtm

**Pull request merged**